jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -143,21 +144,168 @@ ecma_is_constructor (ecma_value_t value) /**< ecma value */
           && ecma_object_is_constructor (ecma_get_object_from_value (value)));
 } /* ecma_is_constructor */
 
+/**
+ * Helper method to count and convert the arguments for the Function/GeneratorFunction constructor call.
+ *
+ * See also:
+ *          ECMA 262 v5.1 15.3.2.1 steps 5.a-d
+ *          ECMA 262 v6 19.2.1.1.1 steps 8
+ *
+ * @return ecma value - concatenated arguments as a string.
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_string_t *
+ecma_op_create_dynamic_function_arguments_helper (const ecma_value_t *arguments_list_p, /**< arguments list */
+                                                  ecma_length_t arguments_list_len) /**< number of arguments */
+{
+  JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
+
+  if (arguments_list_len <= 1)
+  {
+    return ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+  }
+
+  ecma_string_t *str_p = ecma_op_to_string (arguments_list_p[0]);
+
+  if (JERRY_UNLIKELY (str_p == NULL))
+  {
+    return str_p;
+  }
+
+  if (arguments_list_len == 2)
```

Comment:
Is there a test case for this `if`? For some reason it's bit awkward for me. I don't know at the moment which case is this.

Reply:
if the `arguments_list_len == 2` it means a builtin function ctor is called like this: `Function('arg1', 'body')`, so no comma separator is needed after the first argument.  I see no corresponding test so I'll add a new tc for it.